### ISSUE TITLE
Python3 related issues installing glances on Ubuntu 24.04

### ISSUE THREAD

**Issue by maravento** [nicolargo/glances#2737]
**SO** 
Ubuntu 24.04

**Describe the bug**
SyntaxWarning for `python3-pysmi`, `python3-pysnmp4`, `python3-influxdb`when installing glances

**To Reproduce**

```
sudo apt install glances
Leyendo lista de paquetes... Hecho
Creando árbol de dependencias... Hecho
Leyendo la información de estado... Hecho
Se instalarán los siguientes paquetes adicionales:
  fonts-lyx isympy-common isympy3 libjs-jquery-ui liblbfgsb0 libqhull-r8.0
  python-matplotlib-data python3-appdirs python3-bottle python3-bs4 python3-contourpy
  python3-cssselect python3-cycler python3-decorator python3-defusedxml python3-docker
  python3-fonttools python3-fs python3-html5lib python3-influxdb python3-kiwisolver
  python3-lxml python3-lz4 python3-matplotlib python3-mpmath python3-msgpack
  python3-numpy python3-packaging python3-ply python3-pyasn1 python3-pycryptodome
  python3-pysmi python3-pysnmp4 python3-pystache python3-scipy python3-soupsieve
  python3-sympy python3-ufolib2 python3-ujson python3-unicodedata2
  python3-webencodings python3-websocket unicode-data
Paquetes sugeridos:
  glances-doc python3-pynvml libjs-jquery-ui-docs python-cycler-doc python3-genshi
  python-lxml-doc cm-super-minimal dvipng ffmpeg fonts-staypuft inkscape ipython3
  python3-cairocffi python3-gobject python3-pyqt5 python3-sip python3-tornado
  texlive-extra-utils texlive-latex-extra python-mpmath-doc python3-gmpy2 gfortran
  python3-pytest python-ply-doc python-scipy-doc texlive-fonts-extra python-sympy-doc
Se instalarán los siguientes paquetes NUEVOS:
  fonts-lyx glances isympy-common isympy3 libjs-jquery-ui liblbfgsb0 libqhull-r8.0
  python-matplotlib-data python3-appdirs python3-bottle python3-bs4 python3-contourpy
  python3-cssselect python3-cycler python3-decorator python3-defusedxml python3-docker
  python3-fonttools python3-fs python3-html5lib python3-influxdb python3-kiwisolver
  python3-lxml python3-lz4 python3-matplotlib python3-mpmath python3-msgpack
  python3-numpy python3-packaging python3-ply python3-pyasn1 python3-pycryptodome
  python3-pysmi python3-pysnmp4 python3-pystache python3-scipy python3-soupsieve
  python3-sympy python3-ufolib2 python3-ujson python3-unicodedata2
  python3-webencodings python3-websocket unicode-data
0 actualizados, 44 nuevos se instalarán, 0 para eliminar y 0 no actualizados.
Se necesita descargar 0 B/48,9 MB de archivos.
Se utilizarán 216 MB de espacio de disco adicional después de esta operación.
¿Desea continuar? [S/n] s
Extrayendo plantillas para los paquetes: 100%
Seleccionando el paquete python3-defusedxml previamente no seleccionado.
(Leyendo la base de datos ... 248697 ficheros o directorios instalados actualmente.)
Preparando para desempaquetar .../00-python3-defusedxml_0.7.1-2_all.deb ...
Desempaquetando python3-defusedxml (0.7.1-2) ...
Seleccionando el paquete python3-packaging previamente no seleccionado.
Preparando para desempaquetar .../01-python3-packaging_24.0-1_all.deb ...
Desempaquetando python3-packaging (24.0-1) ...
Seleccionando el paquete python3-ujson:amd64 previamente no seleccionado.
Preparando para desempaquetar .../02-python3-ujson_5.9.0-1build1_amd64.deb ...
Desempaquetando python3-ujson:amd64 (5.9.0-1build1) ...
Seleccionando el paquete glances previamente no seleccionado.
Preparando para desempaquetar .../03-glances_3.4.0.3+dfsg-1_all.deb ...
Desempaquetando glances (3.4.0.3+dfsg-1) ...
Seleccionando el paquete fonts-lyx previamente no seleccionado.
Preparando para desempaquetar .../04-fonts-lyx_2.4.0~RC3-1build4_all.deb ...
Desempaquetando fonts-lyx (2.4.0~RC3-1build4) ...
Seleccionando el paquete isympy-common previamente no seleccionado.
Preparando para desempaquetar .../05-isympy-common_1.12-7_all.deb ...
Desempaquetando isympy-common (1.12-7) ...
Seleccionando el paquete python3-mpmath previamente no seleccionado.
Preparando para desempaquetar .../06-python3-mpmath_1.2.1-3_all.deb ...
Desempaquetando python3-mpmath (1.2.1-3) ...
Seleccionando el paquete python3-sympy previamente no seleccionado.
Preparando para desempaquetar .../07-python3-sympy_1.12-7_all.deb ...
Desempaquetando python3-sympy (1.12-7) ...
Seleccionando el paquete isympy3 previamente no seleccionado.
Preparando para desempaquetar .../08-isympy3_1.12-7_all.deb ...
Desempaquetando isympy3 (1.12-7) ...
Seleccionando el paquete libjs-jquery-ui previamente no seleccionado.
Preparando para desempaquetar .../09-libjs-jquery-ui_1.13.2+dfsg-1_all.deb ...
Desempaquetando libjs-jquery-ui (1.13.2+dfsg-1) ...
Seleccionando el paquete liblbfgsb0:amd64 previamente no seleccionado.
Preparando para desempaquetar .../10-liblbfgsb0_3.0+dfsg.4-1build1_amd64.deb ...
Desempaquetando liblbfgsb0:amd64 (3.0+dfsg.4-1build1) ...
Seleccionando el paquete libqhull-r8.0:amd64 previamente no seleccionado.
Preparando para desempaquetar .../11-libqhull-r8.0_2020.2-6build1_amd64.deb ...
Desempaquetando libqhull-r8.0:amd64 (2020.2-6build1) ...
Seleccionando el paquete python-matplotlib-data previamente no seleccionado.
Preparando para desempaquetar .../12-python-matplotlib-data_3.6.3-1ubuntu5_all.deb ...
Desempaquetando python-matplotlib-data (3.6.3-1ubuntu5) ...
Seleccionando el paquete python3-appdirs previamente no seleccionado.
Preparando para desempaquetar .../13-python3-appdirs_1.4.4-4_all.deb ...
Desempaquetando python3-appdirs (1.4.4-4) ...
Seleccionando el paquete python3-bottle previamente no seleccionado.
Preparando para desempaquetar .../14-python3-bottle_0.12.25-1_all.deb ...
Desempaquetando python3-bottle (0.12.25-1) ...
Seleccionando el paquete python3-soupsieve previamente no seleccionado.
Preparando para desempaquetar .../15-python3-soupsieve_2.5-1_all.deb ...
Desempaquetando python3-soupsieve (2.5-1) ...
Seleccionando el paquete python3-bs4 previamente no seleccionado.
Preparando para desempaquetar .../16-python3-bs4_4.12.3-1_all.deb ...
Desempaquetando python3-bs4 (4.12.3-1) ...
Seleccionando el paquete python3-numpy previamente no seleccionado.
Preparando para desempaquetar .../17-python3-numpy_1%3a1.26.4+ds-6ubuntu1_amd64.deb ...
Desempaquetando python3-numpy (1:1.26.4+ds-6ubuntu1) ...
Seleccionando el paquete python3-contourpy previamente no seleccionado.
Preparando para desempaquetar .../18-python3-contourpy_1.0.7-2build1_amd64.deb ...
Desempaquetando python3-contourpy (1.0.7-2build1) ...
Seleccionando el paquete python3-cycler previamente no seleccionado.
Preparando para desempaquetar .../19-python3-cycler_0.11.0-2_all.deb ...
Desempaquetando python3-cycler (0.11.0-2) ...
Seleccionando el paquete python3-decorator previamente no seleccionado.
Preparando para desempaquetar .../20-python3-decorator_5.1.1-5_all.deb ...
Desempaquetando python3-decorator (5.1.1-5) ...
Seleccionando el paquete python3-websocket previamente no seleccionado.
Preparando para desempaquetar .../21-python3-websocket_1.7.0-1_all.deb ...
Desempaquetando python3-websocket (1.7.0-1) ...
Seleccionando el paquete python3-docker previamente no seleccionado.
Preparando para desempaquetar .../22-python3-docker_5.0.3-1ubuntu1_all.deb ...
Desempaquetando python3-docker (5.0.3-1ubuntu1) ...
Seleccionando el paquete python3-scipy previamente no seleccionado.
Preparando para desempaquetar .../23-python3-scipy_1.11.4-6build1_amd64.deb ...
Desempaquetando python3-scipy (1.11.4-6build1) ...
Seleccionando el paquete python3-ufolib2 previamente no seleccionado.
Preparando para desempaquetar .../24-python3-ufolib2_0.16.0+dfsg1-1_all.deb ...
Desempaquetando python3-ufolib2 (0.16.0+dfsg1-1) ...
Seleccionando el paquete python3-fs previamente no seleccionado.
Preparando para desempaquetar .../25-python3-fs_2.4.16-3_all.deb ...
Desempaquetando python3-fs (2.4.16-3) ...
Seleccionando el paquete python3-lxml:amd64 previamente no seleccionado.
Preparando para desempaquetar .../26-python3-lxml_5.2.1-1_amd64.deb ...
Desempaquetando python3-lxml:amd64 (5.2.1-1) ...
Seleccionando el paquete python3-lz4 previamente no seleccionado.
Preparando para desempaquetar .../27-python3-lz4_4.0.2+dfsg-1build4_amd64.deb ...
Desempaquetando python3-lz4 (4.0.2+dfsg-1build4) ...
Seleccionando el paquete python3-unicodedata2 previamente no seleccionado.
Preparando para desempaquetar .../28-python3-unicodedata2_15.1.0+ds-1build1_amd64.deb ..
.
Desempaquetando python3-unicodedata2 (15.1.0+ds-1build1) ...
Seleccionando el paquete unicode-data previamente no seleccionado.
Preparando para desempaquetar .../29-unicode-data_15.1.0-1_all.deb ...
Desempaquetando unicode-data (15.1.0-1) ...
Seleccionando el paquete python3-fonttools previamente no seleccionado.
Preparando para desempaquetar .../30-python3-fonttools_4.46.0-1build2_amd64.deb ...
Desempaquetando python3-fonttools (4.46.0-1build2) ...
Seleccionando el paquete python3-webencodings previamente no seleccionado.
Preparando para desempaquetar .../31-python3-webencodings_0.5.1-5_all.deb ...
Desempaquetando python3-webencodings (0.5.1-5) ...
Seleccionando el paquete python3-html5lib previamente no seleccionado.
Preparando para desempaquetar .../32-python3-html5lib_1.1-6_all.deb ...
Desempaquetando python3-html5lib (1.1-6) ...
Seleccionando el paquete python3-msgpack previamente no seleccionado.
Preparando para desempaquetar .../33-python3-msgpack_1.0.3-3build2_amd64.deb ...
Desempaquetando python3-msgpack (1.0.3-3build2) ...
Seleccionando el paquete python3-influxdb previamente no seleccionado.
Preparando para desempaquetar .../34-python3-influxdb_5.3.1-6_all.deb ...
Desempaquetando python3-influxdb (5.3.1-6) ...
Seleccionando el paquete python3-kiwisolver previamente no seleccionado.
Preparando para desempaquetar .../35-python3-kiwisolver_1.4.4-2build1_amd64.deb ...
Desempaquetando python3-kiwisolver (1.4.4-2build1) ...
Seleccionando el paquete python3-matplotlib previamente no seleccionado.
Preparando para desempaquetar .../36-python3-matplotlib_3.6.3-1ubuntu5_amd64.deb ...
Desempaquetando python3-matplotlib (3.6.3-1ubuntu5) ...
Seleccionando el paquete python3-ply previamente no seleccionado.
Preparando para desempaquetar .../37-python3-ply_3.11-6_all.deb ...
Desempaquetando python3-ply (3.11-6) ...
Seleccionando el paquete python3-pyasn1 previamente no seleccionado.
Preparando para desempaquetar .../38-python3-pyasn1_0.4.8-4_all.deb ...
Desempaquetando python3-pyasn1 (0.4.8-4) ...
Seleccionando el paquete python3-pycryptodome previamente no seleccionado.
Preparando para desempaquetar .../39-python3-pycryptodome_3.20.0+dfsg-1_amd64.deb ...
Desempaquetando python3-pycryptodome (3.20.0+dfsg-1) ...
Seleccionando el paquete python3-pysmi previamente no seleccionado.
Preparando para desempaquetar .../40-python3-pysmi_0.3.4-1_all.deb ...
Desempaquetando python3-pysmi (0.3.4-1) ...
Seleccionando el paquete python3-pysnmp4 previamente no seleccionado.
Preparando para desempaquetar .../41-python3-pysnmp4_4.4.12-2_all.deb ...
Desempaquetando python3-pysnmp4 (4.4.12-2) ...
Seleccionando el paquete python3-pystache previamente no seleccionado.
Preparando para desempaquetar .../42-python3-pystache_0.6.0-1_all.deb ...
Desempaquetando python3-pystache (0.6.0-1) ...
Seleccionando el paquete python3-cssselect previamente no seleccionado.
Preparando para desempaquetar .../43-python3-cssselect_1.2.0-2_all.deb ...
Desempaquetando python3-cssselect (1.2.0-2) ...
Configurando python3-bottle (0.12.25-1) ...
Configurando python3-lz4 (4.0.2+dfsg-1build4) ...
Configurando python3-defusedxml (0.7.1-2) ...
Configurando python3-unicodedata2 (15.1.0+ds-1build1) ...
Configurando fonts-lyx (2.4.0~RC3-1build4) ...
Configurando python3-pycryptodome (3.20.0+dfsg-1) ...
Configurando python3-ply (3.11-6) ...
Configurando isympy-common (1.12-7) ...
Configurando libqhull-r8.0:amd64 (2020.2-6build1) ...
Configurando python3-webencodings (0.5.1-5) ...
Configurando python3-ujson:amd64 (5.9.0-1build1) ...
Configurando unicode-data (15.1.0-1) ...
Configurando python3-decorator (5.1.1-5) ...
Configurando python3-packaging (24.0-1) ...
Configurando libjs-jquery-ui (1.13.2+dfsg-1) ...
Configurando python3-cycler (0.11.0-2) ...
Configurando python3-kiwisolver (1.4.4-2build1) ...
Configurando python3-html5lib (1.1-6) ...
Configurando python3-numpy (1:1.26.4+ds-6ubuntu1) ...
Configurando python3-pysmi (0.3.4-1) ...
/usr/lib/python3/dist-packages/pysmi/codegen/jsondoc.py:935: SyntaxWarning: invalid esca
pe sequence '\s'
  self.textFilter = kwargs.get('textFilter') or (lambda symbol, text: re.sub('\s+', ' ',
 text))
/usr/lib/python3/dist-packages/pysmi/codegen/pysnmp.py:1152: SyntaxWarning: invalid esca
pe sequence '\s'
  self.textFilter = kwargs.get('textFilter') or (lambda symbol, text: re.sub('\s+', ' ',
 text))
Configurando python3-lxml:amd64 (5.2.1-1) ...
Configurando python3-pyasn1 (0.4.8-4) ...
Configurando python3-contourpy (1.0.7-2build1) ...
Configurando python3-msgpack (1.0.3-3build2) ...
Configurando python3-mpmath (1.2.1-3) ...
Configurando python3-cssselect (1.2.0-2) ...
Configurando python-matplotlib-data (3.6.3-1ubuntu5) ...
Configurando python3-appdirs (1.4.4-4) ...
Configurando python3-soupsieve (2.5-1) ...
Configurando python3-websocket (1.7.0-1) ...
Configurando python3-pystache (0.6.0-1) ...
Configurando liblbfgsb0:amd64 (3.0+dfsg.4-1build1) ...
Configurando python3-sympy (1.12-7) ...
Configurando python3-scipy (1.11.4-6build1) ...
Configurando python3-pysnmp4 (4.4.12-2) ...
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncio/cmdgen.py:61: SyntaxWarning: invalid
 escape sequence '\*'
  """Creates a generator to perform SNMP GET query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncio/cmdgen.py:168: SyntaxWarning: invali
d escape sequence '\*'
  """Creates a generator to perform SNMP SET query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncio/cmdgen.py:275: SyntaxWarning: invali
d escape sequence '\*'
  """Creates a generator to perform SNMP GETNEXT query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncio/cmdgen.py:388: SyntaxWarning: invali
d escape sequence '\*'
  """Creates a generator to perform SNMP GETBULK query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncio/ntforg.py:36: SyntaxWarning: invalid
 escape sequence '\*'
  """Creates a generator to send SNMP notification.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/cmdgen.py:25: SyntaxWarning: invali
d escape sequence '\*'
  """Performs SNMP GET query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/cmdgen.py:138: SyntaxWarning: inval
id escape sequence '\*'
  """Performs SNMP SET query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/cmdgen.py:249: SyntaxWarning: inval
id escape sequence '\*'
  """Performs SNMP GETNEXT query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/cmdgen.py:362: SyntaxWarning: inval
id escape sequence '\*'
  """Performs SNMP GETBULK query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/sync/cmdgen.py:26: SyntaxWarning: i
nvalid escape sequence '\*'
  """Creates a generator to perform one or more SNMP GET queries.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/sync/cmdgen.py:131: SyntaxWarning: 
invalid escape sequence '\*'
  """Creates a generator to perform one or more SNMP SET queries.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/sync/cmdgen.py:237: SyntaxWarning: 
invalid escape sequence '\*'
  """Creates a generator to perform one or more SNMP GETNEXT queries.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/sync/cmdgen.py:422: SyntaxWarning: 
invalid escape sequence '\*'
  """Creates a generator to perform one or more SNMP GETBULK queries.
/usr/lib/python3/dist-packages/pysnmp/hlapi/asyncore/sync/ntforg.py:22: SyntaxWarning: i
nvalid escape sequence '\*'
  """Creates a generator to send one or more SNMP notifications.
/usr/lib/python3/dist-packages/pysnmp/hlapi/twisted/cmdgen.py:29: SyntaxWarning: invalid
 escape sequence '\*'
  """Performs SNMP GET query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/twisted/cmdgen.py:147: SyntaxWarning: invali
d escape sequence '\*'
  """Performs SNMP SET query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/twisted/cmdgen.py:265: SyntaxWarning: invali
d escape sequence '\*'
  """Performs SNMP GETNEXT query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/twisted/cmdgen.py:397: SyntaxWarning: invali
d escape sequence '\*'
  """Performs SNMP GETBULK query.
/usr/lib/python3/dist-packages/pysnmp/hlapi/twisted/ntforg.py:27: SyntaxWarning: invalid
 escape sequence '\*'
  """Sends SNMP notification.
Configurando glances (3.4.0.3+dfsg-1) ...
Creating system user 'glances'.
Created symlink /etc/systemd/system/multi-user.target.wants/glances.service → /usr/lib/s
ystemd/system/glances.service.
Configurando python3-influxdb (5.3.1-6) ...
/usr/lib/python3/dist-packages/influxdb/tests/client_test.py:527: SyntaxWarning: invalid
 escape sequence '\('
  "\(use 'n', 'u', 'ms', 's', 'm' or 'h'\)"
/usr/lib/python3/dist-packages/influxdb/tests/influxdb08/client_test.py:309: SyntaxWarni
ng: invalid escape sequence '\('
  "Invalid time precision is given. \(use 's', 'm', 'ms' or 'u'\)"
/usr/lib/python3/dist-packages/influxdb/tests/influxdb08/client_test.py:453: SyntaxWarni
ng: invalid escape sequence '\('
  "Invalid time precision is given. \(use 's', 'm', 'ms' or 'u'\)"
/usr/lib/python3/dist-packages/influxdb/tests/influxdb08/client_test.py:737: SyntaxWarni
ng: invalid escape sequence '\('
  "'permissions' must be \(readFrom, writeTo\) tuple"
Configurando python3-docker (5.0.3-1ubuntu1) ...
Configurando python3-bs4 (4.12.3-1) ...
Configurando python3-fs (2.4.16-3) ...
Configurando python3-ufolib2 (0.16.0+dfsg1-1) ...
Procesando disparadores para libc-bin (2.39-0ubuntu8) ...
Procesando disparadores para man-db (2.12.0-4build2) ...
Procesando disparadores para menu (2.1.50) ...
Procesando disparadores para fontconfig (2.15.0-1.1ubuntu2) ...
Configurando python3-fonttools (4.46.0-1build2) ...
Configurando isympy3 (1.12-7) ...
Configurando python3-matplotlib (3.6.3-1ubuntu5) ...

```

**Comment by RazCrimson** [comment]
These are issues with the dependencies themselves. The issue is just the presence of some characters in doc strings which would not cause any change in behavior of the libraries. 

Python 3.12 has some changes in regards to invalid escape sequences in string literals. That change causes these warnings to pop up. You can read more in detail about it here: https://stackoverflow.com/a/77531416

Nothing can be done by glances to hide/avoid them.

TLDR: They won't cause any problems, but no way to hide them with Python 3.12. Stick to python 3.11 if you don't want them to show up.

Closing the issue as nothing else can be done about this other than the dependencies fixing the warnings.
